GliaWidgets/SecureConversations/ChatTranscript/SecureConversations.TranscriptModel.swift

```diff
@@ -554,13 +555,14 @@ extension SecureConversations.TranscriptModel {
                     divider: ChatItem(kind: .unreadMessageDivider)
                 )
 
+                self.hasUnreadMessages = messagesWithUnreadCount.unreadCount > 0
                 self.historySection.set(itemsWithDivider)
                 self.action?(.refreshSection(self.historySection.index))
                 self.action?(.scrollToBottom(animated: false))
                 completion(messagesWithUnreadCount.messages)
-                if messagesWithUnreadCount.unreadCount > 0 {
-                    self.markMessagesAsRead()
-                }
+                markMessagesAsRead(
+                    with: messagesWithUnreadCount.unreadCount > 0 && !environment.shouldShowLeaveSecureConversationDialog
```

Comment:
Since you added `hasUnreadMessages`, you can use it here
```suggestion
                    with: hasUnreadMessages && !environment.shouldShowLeaveSecureConversationDialog
```

Reply:
Oops, fixed https://github.com/salemove/ios-sdk-widgets/pull/1140/commits/bae9919affacd85e1c261691ce7dd4d40bf67711